File: template/dayXX.py
from typing import List

day_num = "XX"
day_title = "Title"


def read_input(filename: str) -> List[str]:
    with open(filename) as f:
        content = [x.strip('\n') for x in f.readlines()]
    return content


def part1():
    print("Part 1")


def part2():
    print("Part 2")


def main():
    print(f"Day {day_num}: {day_title}")
    part1()
    part2()


if __name__ == '__main__':
    main()

File: template/dayXX.ts
import * as fs from 'fs';

const dayNum: string = "XX";
const dayTitle: string = "Title";

function readInputSync(filename: string): string[] {
    return fs.readFileSync(filename, "utf-8")
        .trimRight().split(/\r?\n/);
}

function part1() {
    console.log('Part 1');
}

function part2() {
    console.log('Part 2');
}

function main() {
    console.log(`Day ${dayNum} : ${dayTitle}`);
    part1();
    part2();
}

main();
